napari/plugins/manager.py

```diff
@@ -0,0 +1,345 @@
+        module_name : str
+            The importable module name
+
+        Raises
+        ------
+        PluginImportError
+            If an error is raised when trying to import `module_name`
+        PluginRegistrationError
+            If an error is raised when trying to register the plugin (such as
+            a PluginValidationError.)
+        """
+        try:
+            mod = importlib.import_module(module_name)
+        except Exception as exc:
+            raise PluginImportError(plugin_name, module_name) from exc
+        try:
+            # prevent double registration (e.g. from entry_points)
+            if self.is_registered(mod):
+                return
+            self.register(mod, name=plugin_name)
+        except Exception as exc:
+            raise PluginRegistrationError(plugin_name, module_name) from exc
+
+
+def entry_points_for(
+    group: str,
+) -> Generator[importlib_metadata.EntryPoint, None, None]:
+    """Yield all entry_points matching "group", from any distribution.
+
+    Distribution here refers more specifically to the information in the
+    dist-info folder that usually accompanies an installed package.  If a
+    package in the environment does *not* have a ``dist-info/entry_points.txt``
+    file, then in will not be discovered by this function.
+
+    Note: a single package may provide multiple entrypoints for a given group.
+
+    Parameters
+    ----------
+    group : str
+        The name of the entry point to search.
+
+    Yields
+    -------
+    Generator[importlib_metadata.EntryPoint, None, None]
+        [description]
+
+    Example
+    -------
+    >>> list(entry_points_for('napari.plugin'))
+    [EntryPoint(name='napari-reg', value='napari_reg', group='napari.plugin'),
+     EntryPoint(name='myplug', value='another.module', group='napari.plugin')]
+
+    """
+    for dist in importlib_metadata.distributions():
+        for ep in dist.entry_points:
+            if ep.group == group:
+                yield ep
+
+
+def modules_starting_with(prefix: str) -> Generator[str, None, None]:
+    """Yield all module names in sys.path that begin with `prefix`.
+
+    Parameters
+    ----------
+    prefix : str
+        The prefix to search
+
+    Yields
+    -------
+    module_name : str
+        Yields names of modules that start with prefix
+
+    """
+    for finder, name, ispkg in pkgutil.iter_modules():
+        if name.startswith(prefix):
+            yield name
+
+
+# regex to parse importlib_metadata.EntryPoint.value strings
+# entry point format:  "name = module.with.periods:attr [extras]"
+entry_point_pattern = re.compile(
+    r'(?P<module>[\w.]+)\s*'
+    r'(:\s*(?P<attr>[\w.]+))?\s*'
+    r'(?P<extras>\[.*\])?\s*$'
+)
+
+
+def iter_plugin_modules(
+    prefix: Optional[str] = None, group: Optional[str] = None
+) -> Generator[Tuple[str, str], None, None]:
+    """Discover unique plugins using naming convention and/or entry points.
```

Comment:
I feel like I wrote here that "unique" is confusing and should be removed, but maybe I just thought it and never wrote it... The explanation below is sufficient.

```suggestion
    """Discover plugins using naming convention and/or entry points.
```